<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data processing</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">omics-grouping-training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="1_data.html">(1) Data</a>
</li>
<li>
  <a href="2_data_processing.html">(2) Data processing</a>
</li>
<li>
  <a href="3_qa.html">(3) QA</a>
</li>
<li>
  <a href="4_stats.html">(4) Differentially abundant molecules - statistical analysis</a>
</li>
<li>
  <a href="5_grouping.html">(5) Grouping</a>
</li>
<li>
  <a href="6_more_info.html">(6) More info</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Data processing</h1>

</div>


<div id="metabolomics-data-matrices" class="section level1">
<h1>Metabolomics data matrices</h1>
<p>Metabolomics data are typically analysed as a matrix of intensities
where one axis consists of the <strong>samples</strong> that were
analysed and the other axis consists of the <strong>features</strong>.
The features of interest for this experiment are the <em>m/z</em> at a
specific retention time (RT).</p>
<p><img src="feature_matrix.png" width="400"
alt="Fig 1.3 Feature X sample matrix of intensity" />  </p>
<p>The entire data processing, from raw mass spectrometry data to a
usable omics data matrix for grouping is beyond the scope of this
tutorial. Instead, we will work with a few key data matrices undergoing
standard metabolomics data processing.</p>
<p>In brief, the following steps were already performed:</p>
<ol style="list-style-type: decimal">
<li><strong>Data reduction</strong> - Full scan (MS1) “peak picking”,
grouping and any retention time correction performed by XCMS software
<span class="citation">(<a href="#ref-smith2006xcms">Smith et al.
2006</a>)</span>.</li>
<li><strong>Quality Control (QC) assessment</strong> - for the removal
of poor-quality metabolites and samples.</li>
<li><strong>Xenobiotic metabolite filter</strong> - for the removal of
metabolites that are either the test substance or biotransformation
products of the test substance <span class="citation">(<a
href="#ref-bowen2023simultaneously">Bowen et al. 2023</a>)</span>.</li>
<li><strong>Identification and removal of features</strong> - blank
filter and missing value filter by QC (≥ 30% missing values in the QC
samples are removed).</li>
<li><strong>Signal-drift and batch-eﬀect correction</strong> - to
account for any signal drift or batch eﬀects <span class="citation">(<a
href="#ref-kirwan2013characterising">Kirwan et al.
2013</a>)</span>.</li>
<li><strong>Further identification and removal of features</strong> -
further filtering of features based on threshold of missing values and
relative standard deviation of the features.</li>
<li><strong>Probabilistic Quotient Normalisation (PQN)</strong> -
accounts for the dilution of complex biological mixtures - correction
factor calculated from the QC samples <span class="citation">(<a
href="#ref-dieterle2006probabilistic">Dieterle et al. 2006</a>; <a
href="#ref-di2016non">Di Guida et al. 2016</a>)</span>.</li>
<li><strong>Missing value imputation</strong> - using k nearest
neighbour approach (for multivariate analysis only) <span
class="citation">(<a href="#ref-hrydziuszko2012missing">Hrydziuszko and
Viant 2012</a>; <a href="#ref-di2016non">Di Guida et al.
2016</a>)</span>.</li>
<li><strong>Generalised logarithm transform (glog)</strong> – a
variance-stabilising transformation that also reduces the weighting of
very intense features in multivariate analysis so that they do not
over-power less intense features that might still be significant (for
multivariate analysis only) <span class="citation">(<a
href="#ref-parsons2007improved">Parsons et al. 2007</a>; <a
href="#ref-di2016non">Di Guida et al. 2016</a>)</span>.</li>
</ol>
<p>The mass spectrometry data processing workflow was developed by the
University of Birmingham metabolomics research groups and the Phenome
Centre Birmingham (PCB). The current implementation uses the R packages
<a
href="https://bioconductor.org/packages/release/bioc/html/struct.html">struct</a>,
<a
href="https://www.bioconductor.org/packages/release/bioc/html/structToolbox.html">structToolbox</a>,
and <a
href="https://bioconductor.org/packages/release/bioc/html/pmp.html">pmp</a>
developed by the PCB informatics team <span class="citation">(<a
href="#ref-lloyd2020struct">Lloyd, Jankevics, and Weber
2020</a>)</span>.</p>
<p>The full details of the data processing methods are found in the <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10944399/bin/204_2024_3680_MOESM1_ESM.pdf">supplemental
materials of the Cefic MATCHING publication</a> <span
class="citation">(<a href="#ref-viant2024demonstrating">Viant et al.
2024</a>)</span>.</p>
<p>The HILIC_POS assay for male rodents was used as an example for this
practical with the following relevant data files for the training
available on the github site:</p>
<ul>
<li><strong><a
href="https://raw.githubusercontent.com/michabohealthscience/omics-grouping-training/main/data/HILIC_POS_male/1_unfiltered.csv">data/HILIC_POS_male/1_unfiltered.csv</a></strong>
- unfiltered intensity matrix following step 1</li>
<li><strong><a
href="https://raw.githubusercontent.com/michabohealthscience/omics-grouping-training/main/data/HILIC_POS_male/2_filtered.csv">data/HILIC_POS_male/2_filtered.csv</a></strong>
- filtered (both samples and features) intensity matrix following steps
1-6</li>
<li><strong><a
href="https://raw.githubusercontent.com/michabohealthscience/omics-grouping-training/main/data/HILIC_POS_male/3_pqn.csv">data/HILIC_POS_male/3_pqn.csv</a></strong>
- PQN normalised intensity matrix following steps 1-7</li>
<li><strong><a
href="https://raw.githubusercontent.com/michabohealthscience/omics-grouping-training/main/data/HILIC_POS_male/4_mv_imputed.csv">data/HILIC_POS_male/4_mv_imputed.csv</a></strong>
- missing value imputed intensity matrix following steps 1-8</li>
<li><strong><a
href="https://raw.githubusercontent.com/michabohealthscience/omics-grouping-training/main/data/HILIC_POS_male/5_glog.csv">data/HILIC_POS_male/5_glog.csv</a></strong>
- glog intensity matrix following steps 1-9</li>
</ul>
<div id="intensity-data-matrix" class="section level2">
<h2>Intensity data matrix</h2>
<p>Let’s start with the pre-filtering and pre-normalisation dataset and
see what a typical metabolomics data matrix looks like.</p>
<pre class="r"><code>hilic_pos_all &lt;- read.csv(file.path(root, &#39;data/HILIC_POS_male/1_unfiltered.csv&#39;), row.names = 1)</code></pre>
<p>Again, we can use the RStudio <strong>Environment</strong> panel and
click on the <code>hilic_pos_all</code> data object to see a more user
friendly view of the data.</p>
<p>The row names refers to the <em>m/z</em> and retention time of a
feature whilst the column names are the samples.</p>
<p>Let’s check how many samples and features we have for this data
matrix, we can use the functions <code>nrow</code> and <code>ncol</code>
for this.</p>
<pre class="r"><code># feature count
nrow(hilic_pos_all)
# sample count
ncol(hilic_pos_all)</code></pre>
<pre><code>## [1] 12123
## [1] 112</code></pre>
<p>Now that we have a better understanding of metabolomics data
matrices, we can briefly explain the quality assessments before getting
started on the statistical analysis.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-bowen2023simultaneously" class="csl-entry">
Bowen, Tara J, Andrew D Southam, Andrew R Hall, Ralf JM Weber, Gavin R
Lloyd, Ruth Macdonald, Amanda Wilson, Amy Pointon, and Mark R Viant.
2023. <span>“Simultaneously Discovering the Fate and Biochemical Effects
of Pharmaceuticals Through Untargeted Metabolomics.”</span> <em>Nature
Communications</em> 14 (1): 4653.
</div>
<div id="ref-di2016non" class="csl-entry">
Di Guida, Riccardo, Jasper Engel, J William Allwood, Ralf JM Weber,
Martin R Jones, Ulf Sommer, Mark R Viant, and Warwick B Dunn. 2016.
<span>“Non-Targeted UHPLC-MS Metabolomic Data Processing Methods: A
Comparative Investigation of Normalisation, Missing Value Imputation,
Transformation and Scaling.”</span> <em>Metabolomics</em> 12: 1–14.
</div>
<div id="ref-dieterle2006probabilistic" class="csl-entry">
Dieterle, Frank, Alfred Ross, Götz Schlotterbeck, and Hans Senn. 2006.
<span>“Probabilistic Quotient Normalization as Robust Method to Account
for Dilution of Complex Biological Mixtures. Application in 1H NMR
Metabonomics.”</span> <em>Analytical Chemistry</em> 78 (13): 4281–90.
</div>
<div id="ref-hrydziuszko2012missing" class="csl-entry">
Hrydziuszko, Olga, and Mark R Viant. 2012. <span>“Missing Values in Mass
Spectrometry Based Metabolomics: An Undervalued Step in the Data
Processing Pipeline.”</span> <em>Metabolomics</em> 8: 161–74.
</div>
<div id="ref-kirwan2013characterising" class="csl-entry">
Kirwan, JA, DI Broadhurst, RL Davidson, and MR Viant. 2013.
<span>“Characterising and Correcting Batch Variation in an Automated
Direct Infusion Mass Spectrometry (DIMS) Metabolomics Workflow.”</span>
<em>Analytical and Bioanalytical Chemistry</em> 405: 5147–57.
</div>
<div id="ref-lloyd2020struct" class="csl-entry">
Lloyd, Gavin Rhys, Andris Jankevics, and Ralf J M Weber. 2020.
<span>“<span class="nocase">struct: an R/Bioconductor-based framework
for standardized metabolomics data analysis and beyond</span>.”</span>
<em>Bioinformatics</em> 36 (22-23): 5551–52. <a
href="https://doi.org/10.1093/bioinformatics/btaa1031">https://doi.org/10.1093/bioinformatics/btaa1031</a>.
</div>
<div id="ref-parsons2007improved" class="csl-entry">
Parsons, Helen M, Christian Ludwig, Ulrich L Günther, and Mark R Viant.
2007. <span>“Improved Classification Accuracy in 1-and 2-Dimensional NMR
Metabolomics Data Using the Variance Stabilising Generalised Logarithm
Transformation.”</span> <em>BMC Bioinformatics</em> 8: 1–16.
</div>
<div id="ref-smith2006xcms" class="csl-entry">
Smith, Colin A, Elizabeth J Want, Grace O’Maille, Ruben Abagyan, and
Gary Siuzdak. 2006. <span>“XCMS: Processing Mass Spectrometry Data for
Metabolite Profiling Using Nonlinear Peak Alignment, Matching, and
Identification.”</span> <em>Analytical Chemistry</em> 78 (3): 779–87.
</div>
<div id="ref-viant2024demonstrating" class="csl-entry">
Viant, Mark R, E Amstalden, T Athersuch, M Bouhifd, S Camuzeaux, DM
Crizer, P Driemert, et al. 2024. <span>“Demonstrating the Reliability of
in Vivo Metabolomics Based Chemical Grouping: Towards Best
Practice.”</span> <em>Archives of Toxicology</em>, 1–13.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
